src/main/java/knightminer/knightperipherals/util/LuaTimer.java

```diff
@@ -35,9 +35,13 @@ public void incrementTimers(ServerTickEvent event) {
 				Entry<IComputerAccess, Integer[]> timer = iterator.next();
 				Integer[] data = timer.getValue();
 				data[0] -= 1;
-				if (data[0] < 1) { // check less than rather than equals just in
-				                   // case
-					timer.getKey().queueEvent(TIMER_EVENT, new Object[] { TIMER_TOKEN + data[1] });
+				if (data[0] < 1) { // check less than rather than equals just in case
+					try {
+						timer.getKey().queueEvent(TIMER_EVENT, new Object[] { TIMER_TOKEN + data[1] });
+					}
+					catch (RuntimeException e) { // Peripheral detached after timer started
+						continue; // Ignore exception
```

Comment:
I'm not a big fan of catching and ignoring runtime exceptions, perhaps we should log the exception?

Reply:
Catching the exception seemed like the easiest solution without rewriting how the timers worked since there's no easy way (from what I gathered) to check if our peripheral is still attached to a valid turtle before we call `queueEvent`. It will just throw if we're not. I didn't see anything else down that code path that should throw, so I don't believe we'd be catching (and ignoring) any other exceptions.

That said, I'd be happy to log it instead. How are you logging things currently? I haven't had much time to dig through the rest of your code.

Reply:
Nvm. I figured it out. Let me know if that looks alright.